AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -187,36 +185,47 @@ module {:options "/functionSyntax:4" } Mutations {
     var mutationLockUUID :- mutationLockUUID?
     .MapFailure(e => Types.KeyStoreAdminException(message := "Could not generate UUID for Mutation Lock. " + e));
 
-    var customEncryptionContext
+    var inferredOriginalEC
       := map k <- activeItem.EncryptionContext
                   // This pull everything that is not in our restricted list.
              | k !in Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES
       :: k := activeItem.EncryptionContext[k];
 
-    var prefixedTerminalCustomEC?: Option<KeyStoreTypes.EncryptionContextString> := None;
+    // IA-BLOCKER :: To Preserve Unexpected/un-modeled Attributes.
+    // We need to copy them from inferredOriginalEC to terminalEC.
+    // Which means we need to select those members without a prefix,
+    // and copy them over to terminal.
+    var unexpectedEC := PrefixUtils.FilterMapForKeysThatDoNotBeginWithPrefix(
+      prefix := Structure.ENCRYPTION_CONTEXT_PREFIX,
+      aMap := inferredOriginalEC
+    );
+    assert unexpectedEC.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES;
+
+    var terminalEC?: Option<KeyStoreTypes.EncryptionContextString> := None;
     if (input.mutations.terminalEncryptionContext.Some?) {
 
-      var prefixedTerminalCustomEC := PrefixUtils.AddingPrefixToKeysOfMapDoesNotCreateCollisions(
+      var terminalEC := PrefixUtils.AddingPrefixToKeysOfMapDoesNotCreateCollisions(
         prefix := Structure.ENCRYPTION_CONTEXT_PREFIX,
         aMap := input.mutations.terminalEncryptionContext.value
-      );
+      ) + unexpectedEC;
       :- Need(
-        prefixedTerminalCustomEC.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES,
+        terminalEC.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES,
         Types.KeyStoreAdminException(message:="Terminal Encryption Context contains a reserved word!")
       );
```

Comment:
it would be nice to have a lemma to prove that they the two things you are adding are disjoint, this way you can remove the Need and replace with an assert. I'd be nice to avoid a runtime check

Reply:
Added a TODO in a2a28ea23